src/site/asn1.rst

```diff
@@ -68,6 +68,43 @@ country ``name`` (string), ``population`` (int), ``area`` (double), and ``contin
 Part 2 --- Country Catalogue
 ============================
 
+The ``CountryCatalogue`` class will be a collection of ``Country`` objects. There will be three fields
+
+    * ``size``, an int to keep track of the number of countries in the catalogue
+    * ``catalogue``, an array of ``Country`` objects
+    * ``countryContinent``, a *hash map* to keep track of the country/continent relationships
+        * A map is the same idea as a dictionary from Python
+        * `How does one use a hash map? <https://www.google.com/search?q=java+hashmap&oq=java+hashmap>`_
+
+There will also be a static constant int for this class called ``DEFAULT_CAPACITY`` that should be set to 10. This will
+be used as a starting size for the ``catalogue`` array.
+
+#. Create a constructor that will setup the catalogue of countries based on some text files --- you may find this to be the must difficult part of the assignment
```

Comment:
Don't instill bad habits. Constructors shouldn't do much work. Make this a function which invokes the default constructor of the catalogue, then uses `add`, etc.

Reply:
I'll switch it out for some method called `loadFromFiles` or something to call after instance exists.

Reply:
It'll still me a method for the `CountryCatalogue`, just so it's in the class and I can keep the `Asn` class where `main` is unadulterated

Reply:
Actually, idk, maybe this will be *Part 3*. It's also probably the toughest part of the assignment. I will just give them an empty static method in `Asn1`. Idk, I'll think about it.

Reply:
> It'll still be a method for the `CountryCatalogue`, just so it's in the class and I can keep the `Asn` class where `main` is unadulterated. This will help it stay aligned with Asn4 from 161, which this is intended to be the _same_, just with Java.

If you're married to it being a single class, it's not unreasonable to have a static factory method for it. Something like:

```java
static CountryCatalogue loadFrom(Path path) throws IOException {
    /* ... call default ctor, add a bunch of times, etc ... */
}
```